gpflow/posterior.py

```diff
@@ -46,19 +63,117 @@ def __init__(self, q_mu, q_sqrt):
         self.q_sqrt = q_sqrt  # [L, M, M], lower-triangular
 
 
-class BasePosterior(Module, ABC):
-    def __init__(self, kernel, inducing_variable, q_mu, q_sqrt, whiten=True, mean_function=None):
+class AbstractPosterior(Module, ABC):
+    def __init__(
+        self,
+        kernel,
+        inducing_variable,
+        q_mu,
+        q_sqrt,
```

Comment:
Taking into account `DeltaDist` signature, this argument should be options with default `None`.

Reply:
I was thinking being explicit and requiring `None` to be passed might be better. Adding type-hints, this would be
```suggestion
        q_sqrt: Union[None, TensorType],
```

Reply:
or we can move to explicitly passing in one of the DeltaDist/DiagDist/... classes